spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/mapping/FirestorePersistentPropertyImpl.java

```diff
@@ -61,6 +61,9 @@ public boolean isIdProperty() {
 	}
 
 	public String getFieldName() {
+		if (isIdProperty()) {
+			return FieldPath.documentId().toString();
```

Comment:
I'd rather make `NAME_FIELD` constant public and reuse it here (yep, it will need its own Javadoc then). No reason do do this dynamically for every field.